### PR TITLE
Export publication rename

### PR DESCRIPTION
**Pull request by anderjo4**

### CODE REVIEW

File: cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishResource.java [hunk 1]
```diff
@@ -259,10 +260,34 @@ public void write(OutputStream os) throws IOException, WebApplicationException {
 		};
 		
 		return Response.ok(stream, MediaType.APPLICATION_OCTET_STREAM)
-				.header("Content-Disposition", "attachment; filename=" + storageFactory.getNameBase(itemId, null) + ".zip")
+				.header("Content-Disposition", "attachment; filename=" + getFilename(items, publication, releaseItem) + ".zip")
 				.build();
 	}
 	
+	private String getFilename(List<CmsItem> items, String publication, CmsItem releaseItem) {
```
Comment (takesson) [inline cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishResource.java]: getFilenameDownload

File: cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishResource.java [hunk 2]
```diff
@@ -259,10 +260,34 @@ public void write(OutputStream os) throws IOException, WebApplicationException {
...
+		long rev = 0;
+		for (CmsItem item: items) {
```
Comment (takesson) [inline cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishResource.java]: Separate method, getRevLatest

File: cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishResource.java [hunk 3]
```diff
@@ -48,6 +48,7 @@
 import se.simonsoft.cms.item.CmsItem;
 import se.simonsoft.cms.item.CmsItemId;
 import se.simonsoft.cms.item.CmsRepository;
+import se.simonsoft.cms.item.RepoRevision;
```
Comment (takesson) [inline cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishResource.java]: ?

File: cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishResource.java [hunk 4]
```diff
@@ -259,10 +260,34 @@ public void write(OutputStream os) throws IOException, WebApplicationException {
...
+			long number = item.getId().getPegRev();
+			if (rev < number) {
+				rev = number;
+			}
+		}
+		
+		String releaseLabel = new CmsItemPublish(releaseItem).getReleaseLabel();
+		
+		if (releaseLabel == null) {
+			throw new IllegalStateException("Item must be a release");
```
Comment (takesson) [inline cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishResource.java]: "The Release does not have a Release Label property."

File: cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishResource.java [hunk 5]
```diff
@@ -259,10 +260,34 @@ public void write(OutputStream os) throws IOException, WebApplicationException {
...
+	private String getFilename(List<CmsItem> items, String publication, CmsItem releaseItem) {
+		long rev = 0;
+		for (CmsItem item: items) {
+			long number = item.getId().getPegRev();
+			if (rev < number) {
+				rev = number;
+			}
+		}
+		
+		String releaseLabel = new CmsItemPublish(releaseItem).getReleaseLabel();
```
Comment (takesson) [inline cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishResource.java]: Add validation to CmsItemPublish such that the 3 gettors don't return empty String. 

They should instead throw IllegalState "Property '...' must not be an empty string."

File: cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishResource.java (hunk 1 above)
Comment (anderjo4) [reply cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishResource.java]: fixed

File: cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishResource.java (hunk 2 above)
Comment (anderjo4) [reply cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishResource.java]: fixed

File: cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishResource.java (hunk 3 above)
Comment (anderjo4) [reply cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishResource.java]: removed

File: cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishResource.java (hunk 4 above)
Comment (anderjo4) [reply cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishResource.java]: fixed

File: cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishResource.java (hunk 5 above)
Comment (anderjo4) [reply cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishResource.java]: fixed.

File: cms-publish-config/src/main/java/se/simonsoft/cms/publish/config/item/CmsItemPublish.java [hunk 6]
```diff
@@ -43,15 +43,27 @@ public boolean isTranslation() {
 	}
 	
 	public String getReleaseLabel() {
-		return item.getProperties().getString("abx:ReleaseLabel");
+		String rl = item.getProperties().getString("abx:ReleaseLabel");
+		if (rl.trim().isEmpty()) {
```
Comment (takesson) [inline cms-publish-config/src/main/java/se/simonsoft/cms/publish/config/item/CmsItemPublish.java]: We talked about null check...

**Pull request merged**